### PR TITLE
Libshout intern

### PR DESCRIPTION
**Pull request by daschuer**
This PR adds libshout to the lib folder and uses it, if the systems libshout suffers 
https://bugs.launchpad.net/mixxx/+bug/1833225

This will be also useful if we later switch to libshout-idjc. This way we can provide the patched library to distributions without idjc.

### CODE REVIEW

File: src/engine/sidechain/shoutconnection.cpp
Comment (uklotzde) [inline src/engine/sidechain/shoutconnection.cpp]: The include directives must not change for bundled libraries to allow replacing a bundled library with a (newer) system library!

Comment (daschuer) [reply src/engine/sidechain/shoutconnection.cpp]: This is just a quick hack to be sure the expected header is used. I think we only need to actually merge this if we discover a libshout bug.
For now I think we can use libshour_idjc to enable aac.

**Comment by daschuer** [comment]
@uklotzde this branch is again relevant for fixing https://bugs.launchpad.net/mixxx/+bug/1833225
can you have a look? How is the Fedora situation?

**Comment by daschuer** [comment]
More and more users are effected as the buggy libshout version spreads the world:
https://bugs.launchpad.net/bugs/1853179
Can one have a look?

**Comment by Holzhaus** [comment]
I'll test this tomorrow. Can you also add this to `CMakeLists.txt` or should I open PR for your repo?

**Comment by daschuer** [comment]
I would be happy if you care about the CMake integration. Thank you

File: build/features.py
Comment (Holzhaus) [inline build/features.py]:
```
$ scons -v
SCons by Steven Knight et al.:
	script: v3.1.1.__BUILD__, 2019-08-08 09:09:12, by none on none
	engine: v3.1.1.__BUILD__, 2019-08-08 09:09:12, by none on none
	engine path: ['/usr/lib/python3.8/site-packages/SCons']
Copyright (c) 2001 - 2019 The SCons Foundation

$ scons -j16
scons: Reading SConscript files ...
  File "/data/jan/Projects/mixxx/build/features.py", line 815

    vars.Add('shoutcast_internal', 'Set to 1 to use internal libshout', 1)

                                                                         ^

TabError: inconsistent use of tabs and spaces in indentation
```

File: build/features.py
Comment (Holzhaus) [inline build/features.py]: Why do you check for version 2.3.1 if the issue is only present in 2.4.2 and 2.4.3?

I read through the linked issue and to me it looks like the issue has been fixed in `master` in this commit: https://gitlab.xiph.org/xiph/icecast-libshout/commit/0ac7ed9e84c3871d4427acc1ce59dca5e4af21ef

I'm sure this fix will be present in the next release of libshout. Can we do a check like this: if version >= 2.4.2 and version <= 2.4.3 then use internal libshout by default, else use system libshout?

**Review by Holzhaus — CHANGES_REQUESTED** [review]

**Comment by Holzhaus** [comment]
> I would be happy if you care about the CMake integration. Thank you

I'll do that later when the 2.2 branch has been merged into master again.

File: build/features.py
Comment (Holzhaus) [inline build/features.py]: Trailing whitespace

File: build/features.py
Comment (Holzhaus) [inline build/features.py]: Trailing whitespace

File: build/features.py
Comment (Holzhaus) [inline build/features.py]: trailing whitespace

File: build/features.py
Comment (Holzhaus) [inline build/features.py]: trailing whitespace

File: build/features.py
Comment (Holzhaus) [inline build/features.py]: trailing whitespace

**Comment by Holzhaus** [comment]
I can report that this builds for me using SCons 3.1.1 when applying these patches:

```diff
diff --git a/build/depends.py b/build/depends.py
index 9d5efa2293..1663efb8c2 100644
--- a/build/depends.py
+++ b/build/depends.py
@@ -1304,7 +1304,7 @@ class MixxxCore(Feature):
             'preferences/dialog/dlgprefvinyldlg.ui',
             'preferences/dialog/dlgprefwaveformdlg.ui',
         ]
-        map(Qt.uic(build), ui_files)
+        list(map(Qt.uic(build), ui_files))

         if build.platform_is_windows:
             # Add Windows resource file with icons and such
diff --git a/build/features.py b/build/features.py
index 5fe3d24475..a68d94b31b 100644
--- a/build/features.py
+++ b/build/features.py
@@ -812,7 +812,7 @@ class LiveBroadcasting(Feature):

     def add_options(self, build, vars):
         vars.Add('shoutcast', 'Set to 1 to enable live broadcasting support', 1)
-   vars.Add('shoutcast_internal', 'Set to 1 to use internal libshout', 1)
+        vars.Add('shoutcast_internal', 'Set to 1 to use internal libshout', 1)

     def configure(self, build, conf):
         if not self.enabled(build):
```

I didn't try to actually start a live broadcast since I don't have a server, but since there aren't any actual code changes I guess it will work.

**Comment by uklotzde** [comment]
We need a CMake integration before merging this.

**Comment by Holzhaus** [comment]
This is for 2.2, there is no CMakeLists.txt in the 2.2 branch.

**Comment by Holzhaus** [comment]
Is this a candidate for 2.2.3-rc2? Or will this go into 2.2.4/2.3.0?

**Comment by daschuer** [comment]
We are quite late to merge this in on one hand. On the other hand it is no new code essentially and it fixes broadcasting for users on effected OSs. So I am leaning to merge this. What do others think?

**Comment by Holzhaus** [comment]
Since this should only cause build-time issues (if any), we can do a late merge IMHO. But please fix the review comments first (without fixing the mixed spaces/tabs error, I cannot build mixxx anymore).

File: build/features.py
Comment (daschuer) [reply build/features.py]: Ups, that was a test with my system libshout which is 2.3.1

**Comment by Holzhaus** [comment]
This branch builds and Mixxx starts without issues.  I can't test the libshout integration because I have no server, but since there are no business logic changes this should work. LGTM.

**Comment by Holzhaus** [comment]
Merge?

**Comment by daschuer** [comment]
Yes, but I a biased.

**Comment by uklotzde** [comment]
I have restarted the macOS build. If it passes we could merge it. As far as I understand this should not have any side effects if you don't enable the new option explicitly.

**Comment by daschuer** [comment]
Yes, no side effect on MacOs and Windows.

It should have the side effect Linux, that it defaults to the internal version, if the system provided version suffers the bug.

**Comment by uklotzde** [comment]
Ok, so let's finally merge this and skip waiting for a successful Travis build.

**Pull request merged**
